### ISSUE TITLE
Handling of global videojs.options is an unmitigated disaster

### ISSUE THREAD

**Issue by imbcmdth** [videojs/video.js#4391]
## Description
Overwriting `videojs.options` with a new object basically ruins any hope you may have had to set options. The `Player` component has an `options_` object on the prototype that is shadowed as `videojs.options`. This means that mutations to `videojs.options` are picked up but completely re-assigning the object has the unexpected side-effect of rendering `videojs.options` meaningless making any changes ineffectual.

**Comment by misteroneill** [comment]
> unmitigated disaster
Tell us how you really feel... 😆 

I agree. This seems like a poor design choice and I wish we'd been able to simplify the `options` story for v6.

I think we out to consider supporting getters and setters to help with this issue in v6. Unfortunately, v6 still needs to support IE8 so that might not be possible (or we'd have to accept that this would continue being an unmitigated disaster in that browser).

**Comment by gkatsev** [comment]
> I think we ought to consider supporting getters and setters to help with this issue in v6.

Noooo, getters and setters should be last choice for anything. The original story for this was `setGlobalOptions` and `getGlobalOptions` but for various reasons, it wasn't used.

A potentially easy solution is in the `videojs()` method, merging `videojs.options` with the passed in `options` before creating the Player.

**Comment by misteroneill** [comment]
Yeah, that might be a good solution - merging `videojs.options` on instantiation seems like it would avoid this problem without the need for getters/setters.
